third_party/scsi_mock/engine.cc

```diff
@@ -33,6 +33,13 @@ ScsiToNvmeResponse ScsiToNvme(unsigned char* cmd_buf, unsigned short cmd_len,
   translator::BeginResponse begin_resp =
       translation.Begin(scsi_cmd, buffer, lun);
 
+  if (begin_resp.status == ApiStatus::kFailure) {
+    Print("Incorrect usage of Translation Library API");
+    ScsiToNvmeResponse resp = {.return_code = 0x40, .alloc_len = 0};
+    translation.AbortPipeline();
```

Comment:
Do we actually leave any resources allocated in the case of an early failure?

Reply:
Good catch! In Translation::Begin, the only case of a `ApiStatus::kFailure` occurs before any resource allocation.